src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    }
+
+    if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
```

Comment:
[CODE STYLE] - There's no need to create a new copy of currentState with `{ ...currentState }` on line 19 since you're not mutating the original state object. You can remove this line.